ibc-core/ics24-host/types/src/error.rs

```diff
@@ -9,6 +9,35 @@ use ibc_primitives::prelude::*;
 use ibc_primitives::proto::Error as ProtoError;
 use prost::DecodeError as ProstError;
 
+/// Errors that originate from host implementations.
+#[derive(Debug, Display)]
+pub enum HostError {
+    /// application module error: `{description}`
+    AppModule { description: String },
+    /// invalid data: `{description}`
+    InvalidData { description: String },
+    /// missing data: `{description}`
+    MissingData { description: String },
+    /// failed to update store: `{description}`
+    FailedToUpdateStore { description: String },
+    /// failed to store data: `{description}`
+    FailedToStoreData { description: String },
+    /// failed to retrieve data from store: `{description}`
+    FailedToRetrieveFromStore { description: String },
+    /// failed to parse data: `{description}`
+    FailedToParseData { description: String },
```

Comment:
Do you think it makes sense to shorten the below variant names? I think they'd still be descriptive enough on their own.

```rust
    /// failed to update store: `{description}`
    FailedToStore { description: String },
    /// failed to retrieve data from store: `{description}`
    FailedToRetrieve { description: String },
    /// failed to parse data: `{description}`
    FailedToParse { description: String },
```

Reply:
[3604b3b](https://github.com/cosmos/ibc-rs/pull/1331/commits/3604b3b66a16fc4f971363650e3806f8880c78d3)